src/main/java/com/parasoft/demoapp/config/OpenApiConfig.java

```diff
@@ -84,4 +96,58 @@ public void customise(OpenAPI openApi) {
                 .build();
     }
 
+    @Bean
+    public SchemaPropertyCustomizer schemaPropertyCustomizer(GlobalPreferencesService globalPreferencesService, SpringDocConfigProperties springDocConfigProperties) {
+        return new SchemaPropertyCustomizer(globalPreferencesService, springDocConfigProperties);
+    }
+
+    public static class SchemaPropertyCustomizer implements PropertyCustomizer {
+        private final GlobalPreferencesService globalPreferencesService;
+        private final SpringDocConfigProperties springDocConfigProperties;
+
+        public SchemaPropertyCustomizer(GlobalPreferencesService globalPreferencesService, SpringDocConfigProperties springDocConfigProperties) {
+            this.globalPreferencesService = globalPreferencesService;
+            this.springDocConfigProperties = springDocConfigProperties;
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public Schema<?> customize(Schema schema, AnnotatedType type) {
+            springDocConfigProperties.getCache().setDisabled(false); // Enable cache to avoid reloading the schema
+
+            if (isRegionType(type)) {
+                IndustryType currentIndustry;
+                try {
+                    currentIndustry = globalPreferencesService.getCurrentIndustry();
+                } catch (GlobalPreferencesNotFoundException | GlobalPreferencesMoreThanOneException e) {
+                    // Will not reach here if project is started up successfully
+                    throw new RuntimeException(e);
+                }
+                List<String> regionsForCurrentIndustry = new ArrayList<>();
+                List<String> regionsForOtherIndustries = new ArrayList<>();
+                for(RegionType regionType : RegionType.values()) {
+                    if(regionType.getIndustryType() == currentIndustry) {
+                        regionsForCurrentIndustry.add(regionType.name());
+                    } else {
+                        regionsForOtherIndustries.add(regionType.name());
+                    }
+                }
+                List<String> reorderedRegions = new ArrayList<>(regionsForCurrentIndustry);
+                reorderedRegions.addAll(regionsForOtherIndustries);
+                schema.setEnum(reorderedRegions);
+            }
+            return schema;
+        }
+
+        private boolean isRegionType(AnnotatedType type) {
+            return type != null &&
+                   type.getType() != null &&
+                   type.getType().getTypeName() != null &&
+                   type.getType().getTypeName().equals("[simple type, class com.parasoft.demoapp.model.industry.RegionType]");
```

Comment:
Is there better way to compare the class instead of comparing the class name as a string?